public/javascript/scripts.js

```diff
@@ -0,0 +1,278 @@
+            if(e.key == "Enter") {
+                processInput(globalInput);
+                globalInput = "";
+            }
+            else {
+                globalInput += e.key;
+            }
+        }
+    });
+});
+
+function initialize() {
+    //load the data from the server here josh
+    var fd = [];
+    for(var i = 0; i < 30; i ++) {
+        fd[i] = fakeData;
+    }
+    
+    //instead of this, load the most recent property changes from the server
+    currentAsset = getDefaultAssetProperties(); //also this function could probably be completely removed and moved to the serve maybe
+    
+    defaultAssetRow = $("#assetTableBody").html();
+    
+    updateAssetTable(fd);
+    
+    assetInfoNoChange = "<tr id='row_TITLE'>" + $("#nochange").html() + "</tr>";
+    assetInfoText = "<tr id='row_TITLE'>" + $("#text").html() + "</tr>";
+    assetInfoDate = "<tr id='row_TITLE'>" + $("#date").html() + "</tr>";
+    assetInfoNumber = "<tr id='row_TITLE'>" + $("#number").html() + "</tr>";
+    assetInfoSelect = "<tr id='row_TITLE'>" + $("#selection").html() + "</tr>";
+    
+    $("#nochange").remove();
+    $("#text").remove();
+    $("#date").remove();
+    $("#number").remove();
+    $("#selection").remove();
+    
+    //IMPORTSNANT see change asset info to find the select and INPUT element chagne listener
+    
+    $(".assetTable tr").click(function(){ 
+        $("#" + selectedTableAsset).removeClass('selected');
+        $(this).addClass('selected');
+        selectedTableAsset = $(this).attr('id');
+        
+        console.log("Get server info for this tag then pass it in here");
+        showAssetInfo(fakeData);
+    });
+}
+
+function updateAssetTable(data) {
+    var html = "";
+    var totalHtml = "";
+    
+    data.forEach(function(item, index) {
+        html = defaultAssetRow;
+        html = html.replace(/TAG/g, item.tag + index);
+        html = html.replace(/PARENT/g, item.parent_tag);
+        html = html.replace(/CUSTOMER/g, item.customer);
+        html = html.replace(/RECEIVED/g, item.received);
+        html = html.replace(/CUST_AG/g, item.customer_tag);
+        html = html.replace(/SERIAL/g, item.serial);
+        html = html.replace(/ASSET_TYPE/g, item.asset_type);
+        html = html.replace(/MANUFACTURER/g, item.manufacturer);
+        html = html.replace(/PRODUCT/g, item.product);
+        html = html.replace(/MODEL/g, item.model);
+        html = html.replace(/LOCATION/g, item.location);
+        
+        totalHtml += html;
+    });
+    
+    $("#assetTableBody").html(totalHtml);
+}
+
+function showAssetInfo(asset) {
+    $("#noAssetMessage").hide();
+    var endHtml = "";
+    for (var property in asset) {
+        if (asset.hasOwnProperty(property)) {
+            var html = ""; //the html for a single property yeuah
+            switch(property) {
+                case 'tag' :
+                    html += assetInfoNoChange;
+                    break;
+                case 'customer':
```

Comment:
This switch statement checks for recognized tag names so the program knows what kind of input to have for each tag. If it is not recognized, it defaults to a text input.

For example, 'received' creates a date input field and adds it to the assetInfo div. 

It would be better to have all the definitions of tags with special inputs and stuff on the server, and have them all loaded in.